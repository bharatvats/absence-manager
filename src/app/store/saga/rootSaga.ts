import { all, call, spawn } from 'redux-saga/effects';
import absenceSaga from './absenceSaga';

export default function* rootSaga() {
    const sagas = [
        absenceSaga
    ];

    yield all(sagas.map(saga => spawn(function* () {
        while (true) {
            try {
                yield call(saga);
                break;
            } catch (ex) {
                console.log(ex);
            }
        }
    })));
}