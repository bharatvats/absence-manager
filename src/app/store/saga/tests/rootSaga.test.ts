import { AllEffect, ForkEffect } from 'redux-saga/effects';
import rootSaga from '../rootSaga';

describe('Root saga suite', () => {
    const rootGen: Generator<AllEffect<ForkEffect<void>>, void, unknown> = rootSaga();

    it('should wait for all sagas to be loaded', () => {
        const allDescriptor = rootGen.next().value;
        expect(allDescriptor).toMatchSnapshot();
    });
});