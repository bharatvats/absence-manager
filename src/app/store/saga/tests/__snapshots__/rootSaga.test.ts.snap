// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root saga suite should wait for all sagas to be loaded 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": true,
  "payload": Array [
    Object {
      "@@redux-saga/IO": true,
      "combinator": false,
      "payload": Object {
        "args": Array [],
        "context": null,
        "detached": true,
        "fn": [Function],
      },
      "type": "FORK",
    },
  ],
  "type": "ALL",
}
`;
